File: MoonEngine-Root/Assets/src/main/java/com/greentree/engine/moon/assets/key/DefaultAssetKey.java
package com.greentree.engine.moon.assets.key;

import com.greentree.commons.util.iterator.IteratorUtil;


public record DefaultAssetKey(Iterable<? extends AssetKey> keys) implements AssetKey {
	
	public DefaultAssetKey(AssetKey... keys) {
		this(IteratorUtil.iterable(keys));
	}
	
}

File: MoonEngine/Assets/src/main/java/com/greentree/engine/moon/assets/key/DefaultAssetKey.kt
package com.greentree.engine.moon.assets.key

import com.greentree.commons.util.iterator.IteratorUtil

data class DefaultAssetKey(val keys: Iterable<AssetKey>) : AssetKey {

	constructor(vararg keys: AssetKey) : this(keys.asIterable())

	override fun toString(): String {
		return "DefaultAssetKey(keys=${IteratorUtil.toString(keys)})"
	}
}